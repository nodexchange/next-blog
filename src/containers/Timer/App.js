/**
 *
 * App
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 */

import React, { Component } from 'react';
import { Helmet } from 'react-helmet';
import { Switch, Route } from 'react-router-dom';

// import HomePage from 'containers/HomePage/Loadable';
// import NotFoundPage from 'containers/NotFoundPage/Loadable';
// import Header from 'components/Header';
// import Footer from 'components/Footer';
import {
  About,
  Contact,
  Careers,
  FrequentlyAsked,
  Front,
  Gdpr,
  Legal,
  NavBar,
  NotFound,
  ProjectDetails,
  ProjectPreview,
  Work,
  Services,
} from 'containers'; // eslint-disable-line import/extensions

import { CookiePanel, FooterBar } from 'components';
import config from '../../config';
import './style.scss';

// eslint-disable-next-line react/prefer-stateless-function
class App extends Component {
  render() {
    return (
      <div className="app">
        <Helmet {...config.app.head} />
        <NavBar />
        <CookiePanel />
        <Switch>
          <Route exact path="/" component={Front} />
          { /* Routes */ }
          <Route exact path="/about" component={About} />
          <Route exact path="/contact" component={Contact} />
          <Route exact path="/careers" component={Careers} />
          <Route exact path="/faq" component={FrequentlyAsked} />
          <Route exact path="/gdpr" component={Gdpr} />
          <Route path="/preview" component={ProjectPreview} />
          <Route exact path="/privacy" component={Legal} />
          <Route exact path="/work" component={Work} />
          <Route path="/work/:projectName" component={ProjectDetails} />
          <Route exact path="/services" component={Services} />
          { /* Catch all route */ }
          <Route path="*" component={NotFound} status={404} />
        </Switch>
        <FooterBar />
      </div>
    );
  }
}

export default App;
