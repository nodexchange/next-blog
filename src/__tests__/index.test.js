/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import App from '../pages/index.js'
import MockData from './mocks/data.mock'

describe('Pages/index.js', () => {
  it('Title prop should match the render', () => {
    const wrap = mount(<App title="Hello Next.js" shows={[]} />)
    expect(wrap.find('#main').text()).toBe('Hello Next.js')
  })
  // it('App shows "Hello world!"', () => {
  //   // console.log(MockData);
  //   const app = shallow(<App shows={MockData} />)
  //   // console.log(app.find('a').text());
  //   // console.log;
  //     // app.contains('a').to.be(true)
  //   expect(app.find('Link')).toEqual(2);
  //   // expect(app.find('p').text()).toEqual('Hello World!')
  //   // expect(2).toBe(2);
  // })
  it('<Li> should match the shows data', () => {
    // const component = renderer.create(<App />)
    // const tree = component.toJSON()
    const wrap = mount(<App title="Hello Next.js" shows={[]} />)
    // console.log(tree);
    // expect(wrap.find('#main').text()).toBe('Hello Next.js')
    // console.log(wrap.debug());
    expect(wrap.find('Link')).toHaveLength(3);
  
  });
})

describe('With Snapshot Testing', () => {
  it('App shows "Hello world!"', () => {
    const component = renderer.create(<App title="Hello Next.js" shows={[]} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})


