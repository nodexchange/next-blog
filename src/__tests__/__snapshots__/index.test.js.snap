// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Hello world!" 1`] = `
<div
  style={
    Object {
      "border": "1px solid #DDD",
      "margin": 20,
      "padding": 20,
    }
  }
>
  <div>
    <a
      href="/"
      onClick={[Function]}
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      Home
    </a>
    <a
      href="/about"
      onClick={[Function]}
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      About
    </a>
    <a
      href="/next"
      onClick={[Function]}
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      Next
    </a>
  </div>
  <h1>
    Batman TV Shows
  </h1>
  <div
    id="main"
  >
    Hello Next.js
  </div>
  <ul />
</div>
`;
